src/oscar.ml

```diff
@@ -0,0 +1,53 @@
+open Ast
+
+let rec eval = function
+      Id(s) -> s
+    | Int_Lit(x) -> x
+    | Double_Lit(x) -> x
+    | Char_Lit(x) -> x
+    | String_Lit(x) -> x
+    | Bool_Lit(x) -> x
+    | Binop(e1, op, e2) ->
+        let v1 = eval e1 and v2 = eval e2 in
+        (
+          match op with
+              Add -> v1 + v2
+            | Sub -> v1 - v2
+            | Mult -> v1 * v2
+            | Div -> v1 / v2
+            | Mod -> v1 % v2
+            | Equal -> v1 = v2
+            | Neq -> v1 <> v2
+            | Less -> v1 < v2
+            | Leq -> v1 <= v2
+            | Greater -> v1 > v2
+            | Geq -> v1 >= v2
+            | And -> v1 && v2
+            | Or -> v1 || v2
+            | Bit_And -> v1 land v2
+            | Bit_Or -> v1 lor v2
+            | Bit_Xor -> v1 lxor v2
+            | Bit_RShift -> v1 lsr v2
+            | Bit_LShift -> v1 lsl v2
+        )
+    | Unop(op, e) ->
+        let v = eval e in
+        (
+          match op with
+              Neg -> -v
+            | Not -> !v
+            (* todo: i couldn't find Bit_Not in OCaml *)
```

Comment:
Bit_Not is not listed in the Bitwise operations section here https://caml.inria.fr/pub/docs/manual-ocaml/libref/Pervasives.html